### PR TITLE
[IMP] web: improve setting search

### PR DESCRIPTION
**Pull request by jpp-odoo**
This commit aim to improve the setting search, and extend it to any text in a setting tag.

### CODE REVIEW

File: addons/web/static/src/webclient/settings_form_view/settings_form_compiler.js [hunk 1]
```diff
@@ -53,6 +53,7 @@ function compileSetting(el, params) {
     let string = toStringExpression(el.getAttribute("string") || "");
     let addLabel = true;
     params.labels = [];
+    params.labels.push(highlightElement(el));
```
Comment (aab-odoo) [inline addons/web/static/src/webclient/settings_form_view/settings_form_compiler.js:56]:
```suggestion
    params.labels = [highlightElement(el)];
```

Comment (aab-odoo) [reply addons/web/static/src/webclient/settings_form_view/settings_form_compiler.js:56]: shouldn't it be
```js
 params.labels = [...highlightElement(el)];
```
?

File: addons/web/static/src/webclient/settings_form_view/settings_form_compiler.js [hunk 2]
```diff
@@ -103,34 +104,27 @@ function compileLabel(el, params) {
...
-    }
-    return res;
-}
-
-function compileGenericLabel(el, params) {
-    const res = this.compileGenericNode(el, params);
-    if (res.textContent) {
-        params.labels.push(res.textContent.trim());
-        highlightElement(res);
+        params.labels.push(...highlightElement(res));
```
Comment (aab-odoo) [inline addons/web/static/src/webclient/settings_form_view/settings_form_compiler.js:108]: if the label is inside the setting, doesn't it work already thanks to the call line 56?

File: addons/web/static/src/webclient/settings_form_view/settings_form_compiler.js (hunk 1 above)
Comment (jpp-odoo) [reply addons/web/static/src/webclient/settings_form_view/settings_form_compiler.js:56]: yep, bad refactoring ... sorry

File: addons/web/static/src/webclient/settings_form_view/settings_form_compiler.js [hunk 3]
```diff
@@ -3,192 +3,167 @@
...
-        imgurl: toStringExpression(module.imgurl),
-        selectedTab: "settings.selectedTab",
-    });
-
-    for (const child of el.children) {
-        append(settingsApp, this.compileNode(child, params));
+export class SettingsFormCompiler extends FormCompiler {
+    setup() {
+        super.setup();
+        this.compilers.unshift(
```
Comment (aab-odoo) [inline addons/web/static/src/webclient/settings_form_view/settings_form_compiler.js:11]: can be push

File: addons/web/static/src/webclient/settings_form_view/settings_form_compiler.js [hunk 4]
```diff
@@ -3,192 +3,167 @@
...
-    });
-    setting.setAttribute("string", string);
-    setting.setAttribute("addLabel", addLabel);
-    setting.setAttribute("labels", JSON.stringify(params.labels));
-    return setting;
-}
+        while (res.lastChild) {
+            res.removeChild(res.lastChild);
+        }
+        append(res, settingsPage);
```
Comment (aab-odoo) [inline addons/web/static/src/webclient/settings_form_view/settings_form_compiler.js:38]: check this, looks like we compile children twice, could be nice to "simply" move them

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**